src/main/scala/leon/termination/StructuralSize.scala

```diff
@@ -66,10 +88,30 @@ trait StructuralSize {
       case TupleType(argTypes) => argTypes.zipWithIndex.map({
         case (_, index) => size(tupleSelect(expr, index + 1, true))
       }).foldLeft[Expr](InfiniteIntegerLiteral(0))(Plus)
+      case IntegerType =>
+        FunctionInvocation(typedAbsBigIntFun, Seq(expr)) 
       case _ => InfiniteIntegerLiteral(0)
     }
   }
 
+  def lexicographicDecreasing(s1: Seq[Expr], s2: Seq[Expr], strict: Boolean, sizeOfOneExpr: Expr => Expr): Expr = {
+    // Note: The Equal and GreaterThan ASTs work for both BigInt and Bitvector
+    
+    val sameSizeExprs = for ((arg1, arg2) <- (s1 zip s2)) yield Equals(sizeOfOneExpr(arg1), sizeOfOneExpr(arg2))
+    
+    val greaterBecauseGreaterAtFirstDifferentPos =
+      orJoin(for (firstDifferent <- 0 until scala.math.min(s1.length, s2.length)) yield and(
+          andJoin(sameSizeExprs.take(firstDifferent)),
+          GreaterThan(sizeOfOneExpr(s1(firstDifferent)), sizeOfOneExpr(s2(firstDifferent)))
+      ))
+      
+    if (s1.length > s2.length || (s1.length == s2.length && !strict)) {
+      or(andJoin(sameSizeExprs), greaterBecauseGreaterAtFirstDifferentPos)
```

Comment:
Yes, having finite argument list lengths is crucial, because it's needed for the lexicographic order to be well-founded. Otherwise we could have infinite decreasing chains like `b > ab >  aab > aaab ... `

Regarding `reverse_reverse_equivalence`, I also was a bit surprised that it didn't work, but I don't understand the `Strengthener` well enough in order to debug it...